src/main/java/seedu/address/model/CommandHistoryManager.java

```diff
@@ -21,6 +22,9 @@ public CommandHistoryManager() {
     public void saveCommandExecutionString(String commandInputString) {
         //Every execution of a new command will cause the currIndex to be reset
         //to the right-most index of the `history` + 1
+        if (this.history.size() == this.capacity) {
```

Comment:
this is out of context for me so maybe it's a dumb doubt, but when you do history.remove() you are removing the last item in the history and then history.add(commandInputString) is adding it into the last position right?

Reply:
Oh nope, the java documentation says that .remove() removes the element at the head of the linkedlist, not the tail